### ISSUE TITLE
feature request: toggle between site versions (releases)

### ISSUE THREAD

**Issue by medyagh** [mmistakes/minimal-mistakes#1115]
I was wondering if this is do-able ? I wanna be able to toggle my site between versions, like how did the website look like in release 1.0.0 and 2.0.0 is that possible with minimal-mistakes

**Comment by mmistakes** [comment]
Not sure why you'd want to do this, but from my perspective it's probably next to impossible to impliment.

Each major version of the theme can be wildly different between the underlining HTML and CSS. `4.0` is a complete rebuild and almost none of it shares the same code from prior versions.

If it was just CSS changes you could probably rig up something that swaps what `.css` is loaded, but because the actual class names and structure of most `_layouts` and `_includes` change between versions that won't work.

**Comment by medyagh** [comment]
oh I meant the content of my site, like let say I have a api documentation and each time I have a major change in api doc, I want users to go back in time and be able to see the docs exactly the way it was for previous version of the api.

would it be better if I just copy the current site files to a folder ?

**Comment by medyagh** [comment]
so maybe my question could be changed to this, 
can I toggle the site to reffer to a different folder based on a dropdown ? with a default of Latest Folder?

**Comment by mmistakes** [comment]
Probably. I don't have much experience with doing that. I think that's more of a Jekyll question than a theme question.

Might get some better help asking on the [Jekyll Talk forum](https://talk.jekyllrb.com/). You could look at the [Bootstrap repo](https://github.com/twbs/bootstrap) too for ideas. They use Jekyll to build their docs and I'm pretty sure they structure it in a way to archive each major version.

**Comment by medyagh** [comment]
sweet thanks a lot.

**Comment by annegentle** [comment]
I've been working on solving to add versions using Minimal Mistakes and wrote up my findings in this blog post: https://justwriteclick.com/2017/07/30/investigating-jekyll-for-versioned-content/

tl;dr - It was easiest to use a one-off _config.version.yml file for each version with a new baseurl: /n.n setting. I'm still working on the UI menu itself, but one super-simple idea would be to add the versions to the top menu in `_data/navigation.yml` and make each version's URL selectable there.

**Comment by medyagh** [comment]
@annegentle thanks will check it out.
